File: algorithms/nim/Fibonacci.nim
import unittest


proc recursive_fibonacci(n: uint): uint =
    if n < 2:
        result = n
    else:
        result = recursive_fibonacci(n - 1) + recursive_fibonacci(n - 2)

      
suite "recursive_fibonacci":
    test "equals":
        check recursive_fibonacci(0) == 0
        check recursive_fibonacci(1) == 1
        check recursive_fibonacci(2) == 1
        check recursive_fibonacci(3) == 2
        check recursive_fibonacci(4) == 3
        check recursive_fibonacci(5) == 5
        check recursive_fibonacci(6) == 8
        check recursive_fibonacci(7) == 13
        check recursive_fibonacci(8) == 21

File: algorithms/rust/Fibonacci.rs
fn recursive_fibonacci(n: u64) -> u64 {
    if n < 2 {
        n
    } else {
        recursive_fibonacci(n - 1) + recursive_fibonacci(n - 2)
    }
}

#[test]
fn test_recursive_fibonacci() {
    assert_eq!(recursive_fibonacci(0), 0);
    assert_eq!(recursive_fibonacci(1), 1);
    assert_eq!(recursive_fibonacci(2), 1);
    assert_eq!(recursive_fibonacci(3), 2);
    assert_eq!(recursive_fibonacci(4), 3);
    assert_eq!(recursive_fibonacci(5), 5);
    assert_eq!(recursive_fibonacci(6), 8);
    assert_eq!(recursive_fibonacci(7), 13);
    assert_eq!(recursive_fibonacci(8), 21);
}

